core/src/main/java/jenkins/cli/ListNodesCommand.java

```diff
@@ -0,0 +1,572 @@
+        if (minExecutors < 0) {
+            stderr.println("Invalid input: The minimum number of executors (-min-executors) must be a non-negative value.");
+
+            return false;
+        }
+
+        // Validate limit
+        if (limit <= 0) {
+            stderr.println("Invalid input: -limit must be a positive integer.");
+
+            return false;
+        }
+
+        // Validate labels
+        if (labels != null && !labels.trim().isEmpty()) {
+            // Check length to prevent excessive input
+            if (labels.length() > MAX_LABELS_LENGTH) {
+                stderr.println("Invalid input: -labels input exceeds maximum length of " + MAX_LABELS_LENGTH + " characters.");
+
+                return false;
+            }
+
+            if (!parseLabelExpression()) {
+                return false; // label parsing fails, return false
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Parses the label expression from the command-line input.
+     * If the expression is invalid, it prints an error message and returns false.
+     *
+     * @return true if the label expression is valid; false otherwise.
+     */
+    private boolean parseLabelExpression() {
+        try {
+            labelExpression = Label.parseExpression(labels.trim());
+
+            if (labelExpression == null) {
+                stderr.println("Invalid input: The label expression '" + labels + "' is not valid.");
+
+                return false;
+            }
+        } catch (IllegalArgumentException e) {
+            stderr.println("Invalid input: The label expression '" + labels + "' is not valid. " + e.getMessage());
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Excludes the Jenkins controller node if the excludeController option is set.
+     * The controller is excluded only if it is not null and is an instance of Jenkins.
+     *
+     * @param computer The computer to check.
+     * @return true if the computer should be included, false if it should be excluded.
+     */
+    private boolean excludeControllerIfNeeded(Computer computer) {
+        if (!excludeController) {
+            return true;
+        }
+
+        Node node = computer.getNode();
+        if (node == null) {
+            return true;
+        }
+
+        return !(node instanceof Jenkins);
+    }
+
+    /**
+     * Checks if the computer matches the specified status filter.
+     * If the status is ALL, it returns true for all computers.
+     *
+     * @param computer The computer to check.
+     * @return true if the computer matches the status filter, false otherwise.
+     */
+    private boolean matchesStatus(Computer computer) {
+        switch (status) {
+            case TEMP_OFFLINE:
+                return computer.isTemporarilyOffline();
+            case ONLINE:
+                return computer.isOnline();
+            case OFFLINE:
+                return !computer.isOnline();
```

Comment:
negation of a negation https://github.com/jenkinsci/jenkins/blob/c5b876b582ab2e5676670952cbc089ef71aa368d/core/src/main/java/hudson/model/Computer.java#L625
technically nothing wrong with it as I believe the compiler fixes it, but
```suggestion
                return computer.isOffline();
```
reads better.

Reply:
Good point, thanks. I'll update it to use `computer.isOffline()` instead.